server/routes/sockets/room-service.js

```diff
@@ -106,10 +141,15 @@ class RoomService {
             if (!this.activeRoomsByID[eventInfo.roomID]) {
                 this.emitUserEvent('room_inactive', socket);
             } else {
+                // Construct the user object to pass to the room
+                const user = {
+                    nickname: eventInfo.nickname,
+                    socketID: socket.id,
+                }
                 // Grab the correct room from the list of active rooms
                 const room = this.activeRoomsByID[eventInfo.roomID];
                 // Join the user to the room
-                room.joinUserToRoom(eventInfo.nickname, socket);
+                room.joinUserToRoom(user, socket);
```

Comment:
Why does the `socket.id` need to exist in the user object if it's already an attribute of the `socket`? I see later in `room.js` you refactored to use a user object there as well. 

It seems like the room only needs to know the nickname and the socket, so why not keep the parameters matching that structure (`(nickname, socket)`), and offload the responsibility of creating a user object to the room?